File: hw/pazmino/u2/HW21Mongo/MongoDBC/src/main/java/ec/edu/espe/mongodbc/controller/StudentController.java
package controller;

import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoCursor;
import model.Student;
import org.bson.Document;
import utils.MongoDBManager;

import java.util.ArrayList;
import java.util.List;

public class StudentController {
    private MongoDBManager mongoDBManager;
    private MongoCollection<Document> studentCollection;

    public StudentController() {
        mongoDBManager = new MongoDBManager();
        studentCollection = mongoDBManager.getCollection("STUDENTS");
    }

    public void addStudent(Student student) {
        Document document = new Document("id", student.getId())
                .append("name", student.getName())
                .append("signature", student.getSignature())
                .append("career", student.getCareer());
        studentCollection.insertOne(document);
    }

    public List<Student> readStudents() {
        List<Student> students = new ArrayList<>();
        MongoCursor<Document> cursor = studentCollection.find().iterator();

        while (cursor.hasNext()) {
            Document doc = cursor.next();
            Student student = new Student(
                    doc.getString("id"),
                    doc.getString("name"),
                    doc.getString("signature"),
                    doc.getString("career")
            );
            students.add(student);
        }
        return students;
    }

    public void updateStudent(String id, String updatedStudent, String signature, String career) {
        Document query = new Document("id", id);
        Document updatedDocument = new Document("$set", new Document("name", updatedStudent.getName())
                .append("signature", updatedStudent.getSignature())
                .append("career", updatedStudent.getCareer()));
        studentCollection.updateOne(query, updatedDocument);
    }

    public void deleteStudent(String id) {
        Document query = new Document("id", id);
        studentCollection.deleteOne(query);
    }
}

File: hw/espinosa/U2/HWPythonDataBase/controller/student_controller.py
from model.student import Student
from bson import ObjectId

class StudentController:
    def __init__(self, mongodb_manager):
        self.mongodb_manager = mongodb_manager
        self.collection = self.mongodb_manager.get_collection('students')
    
    def add_student(self, name, gpa, age):
        student = Student(name=name, gpa=gpa, age=age)
        result = self.collection.insert_one(student.to_dict())
        return str(result.inserted_id)
    
    def get_all_students(self):
        students = []
        for doc in self.collection.find():
            students.append(Student.from_dict(doc))
        return students
    
    def get_student_by_id(self, student_id):
        try:
            doc = self.collection.find_one({"_id": ObjectId(student_id)})
            if doc:
                return Student.from_dict(doc)
            return None
        except Exception:
            return None
    
    def update_student(self, student_id, name, gpa, age):
        try:
            result = self.collection.update_one(
                {"_id": ObjectId(student_id)},
                {"$set": {"name": name, "gpa": gpa, "age": age}}
            )
            return result.modified_count > 0
        except Exception:
            return False
    
    def delete_student(self, student_id):
        try:
            result = self.collection.delete_one({"_id": ObjectId(student_id)})
            return result.deleted_count > 0
        except Exception:
            return False